pool.js

```diff
@@ -309,7 +323,7 @@ Pool.prototype.add_endpoint = function (host_port) {
     var ip = ip_port[0];
     var port = +ip_port[1];
     if (port > 0 && port < 65536) {
-        var endpoint = new GO.PoolEndpoint(this.http, ip, port, this.options);
+        var endpoint = new GO.PoolEndpoint(this.protocol, ip, port, this.options);
```

Comment:
it seems like you intended for `agent` to be a property of `options` but I don't see that anywhere